src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useBlockNumber.tsx

```diff
@@ -0,0 +1,23 @@
+import type { ChainId } from '@shapeshiftoss/caip'
+import { useEffect, useState } from 'react'
+import { assertGetViemClient } from 'lib/viem-client'
+
+export const useBlockNumber = (chainId: ChainId | undefined, enabled: boolean) => {
```

Comment:
Decided we should be triggering checks based on new blocks rather than blindly polling. Keen for feedback on this if it's a flawed approach

Reply:
Very smort - agree that's much better than polling.

Wondering if we could we use https://wagmi.sh/react/hooks/useBlockNumber instead of a custom hook? This would allow us to benefit from the fact that wagmi uses react-query under the hood, and since hooks are not singletons, this would make things less fetchy and faster.

Otherwise conceptually happy, since an allowance obviously cannot change without a new block being mined.

Reply:
Discussed this one out and we don't want to bring back wagmi for `useBlockNumber` only